### PR TITLE
UNI-159 #comment Add button and action

### PR DESCRIPTION
**Pull request by brev**
WIP UNI-159 #comment WIP add button and action

FYI @jcasner @lscheinkman @marionleborgne

### CODE REVIEW

**Comment by brev** [comment]
onClick is not dispatching Action correctly yet

File: unicorn/frontend/browser/actions/add.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+'use strict';
+
+
+/**
+ * Fluxible Action: Add (user adding new file)
+ */
+
+module.exports = (actionContext, payload) => {
+  actionContext.dispatch('ADD_ACTION', payload);
```
Comment (lscheinkman) [inline unicorn/frontend/browser/actions/add.js]: You should return the value returned by `actionContext.dispatch` here.

From http://fluxible.io/api/actions.html
> If the action takes less than three parameters, executeAction will resolve the promise with the return value

File: unicorn/frontend/browser/components/foo.jsx [hunk 2]
```diff
@@ -32,13 +32,21 @@ import React from 'react';
...
+import AddAction from '../actions/add';
+
 let Theme = new Material.Styles.ThemeManager();
 
+let Card = Material.Card;
+let CardHeader = Material.CardHeader;
+let CardText = Material.CardText;
+let FloatingActionButton = Material.FloatingActionButton;
+let FontIcon = Material.FontIcon;
+let LeftNav = Material.LeftNav;
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/foo.jsx]: Just a question of style but the above code could be replaced with:
```javascript
let {
    Card, CardHeader, CardText, FloatingActionButton, FontIcon, LeftNav
} = Material;
```

File: unicorn/frontend/browser/components/foo.jsx [hunk 3]
```diff
@@ -66,14 +83,33 @@ module.exports = React.createClass({
...
     return (
       <div>
-        <LeftNav ref="leftNav" menuItems={menuItems} />
-        <h1>Welcome</h1>
-        <p>{this.props.foo}</p>
+        <Card style={{ marginLeft: '256px' }}>
+          <CardHeader title="Title" subtitle="Subtitle" />
+          <CardText>
+            <h1>Welcome</h1>
+            <FloatingActionButton onClick={this.onClick}>
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/foo.jsx]: You need to manually bind callbacks when using ES6 syntax. See https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html#autobinding
```javascript
<FloatingActionButton onClick={this.onClick.bind(this}>
```

File: unicorn/frontend/browser/actions/add.js (hunk 1 above)
Comment (brev) [inline unicorn/frontend/browser/actions/add.js]: wow, nice catch, thanks!

File: unicorn/frontend/browser/components/foo.jsx (hunk 2 above)
Comment (brev) [inline unicorn/frontend/browser/components/foo.jsx]: :+1: thanks

File: unicorn/frontend/browser/components/foo.jsx (hunk 3 above)
Comment (brev) [inline unicorn/frontend/browser/components/foo.jsx]: thank you! Saved me a LOT of time.

**Comment by brev** [comment]
still in progress

**Comment by lscheinkman** [comment]
@brev I've fixed the events on unicorn with this PR #242 
Let me know if it works for you

**Comment by brev** [comment]
locked and loaded, merging.
FYI @lscheinkman @marionleborgne @jcasner

**Comment by lscheinkman** [comment]
:+1:

**Pull request merged**